warn/warn.go

```diff
@@ -90,7 +94,9 @@ func makeLinterFinding(node build.Expr, message string, replacement ...LinterRep
 
 // RuleWarningMap lists the warnings that run on a single rule.
 // These warnings run only on BUILD files (not bzl files).
-var RuleWarningMap = map[string]func(f *build.File, pkg string, expr build.Expr) *Finding{}
+var RuleWarningMap = map[string]func(call *build.CallExpr, pkg string) *LinterFinding{
+	"positional-args": positionalArgumentsWarning,
+}
```

Comment:
Is it still useful to have a separate map?
(instead of using `RuleWarning(positionalArgumentsWarning)`)

Reply:
Yes, because some rule warnings also require a `pkg` argument, and none of file warning functions need it.

I introduced `RuleWarning(...)` a couple of weeks ago, but now that I've also refactored the internal functions I see that this approach doesn't always work.